### PR TITLE
PR for gridpacks production for monoLQ dark matter analysis

### PR DESCRIPTION
**Pull request by abdollah110**
I have just made one example, but a code is committed (dirGenerator_gridpack.py) which can make scripts to generate mane gridpacks for various signal mass points

### CODE REVIEW

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/DarkMatter_MonoLQ2/Codex_LQ1000_DM_400_X_440_gen2/Codex_LQ1000_DM_400_X_440_gen2_proc_card.dat [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+#************************************************************
+set group_subprocesses Auto
+set ignore_six_quark_processes False
+set loop_optimized_output True
+set loop_color_flows False
+set gauge unitary
+set complex_mass_scheme False
+set max_npoint_for_channel 0
+import model sm
+define p = g u c d s u~ c~ d~ s~
```
Comment (perrozzi) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/DarkMatter_MonoLQ2/Codex_LQ1000_DM_400_X_440_gen2/Codex_LQ1000_DM_400_X_440_gen2_proc_card.dat]: please follow the recommendations in https://twiki.cern.ch/twiki/bin/view/CMS/Moriond18MC#Checks_to_the_datacards_BEFORE_T

**Comment by abdollah110** [comment]
I have updated the proc card as recommended in the twiki.
Please check the proc and run .dat files and let me know if everything is fine

File: bin/MadGraph5_aMCatNLO/Template/_proc_card.dat [hunk 2]
```diff
@@ -0,0 +1,24 @@
+set group_subprocesses Auto
+set ignore_six_quark_processes False
+set loop_optimized_output True
+set loop_color_flows False
+set gauge unitary
+set complex_mass_scheme False
+set max_npoint_for_channel 0
+import model sm
```
Comment (perrozzi) [inline bin/MadGraph5_aMCatNLO/Template/_proc_card.dat]: please remove multiple import statements as only the last one is effective

File: bin/MadGraph5_aMCatNLO/Template/_run_card.dat [hunk 3]
```diff
@@ -0,0 +1,280 @@
+#*********************************************************************
```
Comment (perrozzi) [inline bin/MadGraph5_aMCatNLO/Template/_run_card.dat]: is this file really needed?

File: bin/MadGraph5_aMCatNLO/Template/_customizecards.dat [hunk 4]
```diff
@@ -0,0 +1,13 @@
+set param_card  frblock  1  4.000000e+02
```
Comment (perrozzi) [inline bin/MadGraph5_aMCatNLO/Template/_customizecards.dat]: there are a lot of filenames which are not good...

**Comment by abdollah110** [comment]
But one is for sm while the other is for lq_met
The first one is to define the leptons and the other is to define the
signal process.
So do you suggest to remove the 'import model sm'?


Cheers
Abdollah

File: bin/MadGraph5_aMCatNLO/Template/_customizecards.dat (hunk 4 above)
Comment (abdollah110) [reply bin/MadGraph5_aMCatNLO/Template/_customizecards.dat]: here we define the mass of dark matter. So it is necessary

Comment (perrozzi) [reply bin/MadGraph5_aMCatNLO/Template/_customizecards.dat]: yes but are you sure you really want these info to be stored in a file called
bin/MadGraph5_aMCatNLO/Template/_customizecards.dat ?
probably not...

**Comment by perrozzi** [comment]
yes please remove `import model sm` because is ignored by mg5_amc

**Comment by abdollah110** [comment]
Removed import model sm

File: bin/MadGraph5_aMCatNLO/Template/_customizecards.dat (hunk 4 above)
Comment (abdollah110) [reply bin/MadGraph5_aMCatNLO/Template/_customizecards.dat]: If these are not stored in the _customizecards.dat, then where should they be stored?
I am not expert on MG5, so if you have a better idea, please let me know, otherwise, I see no problem of defining them in another file

**Comment by perrozzi** [comment]
Hi, you are missing the name of the process in the datacards, cfr 
https://twiki.cern.ch/twiki/bin/viewauth/CMS/QuickGuideMadGraph5aMCatNLO#Preparation_of_the_cards
also, some of the files are correctly stored into directories, like
bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/DarkMatter_MonoLQ2/Codex_LQ1000_DM_400_X_440_gen2/Codex_LQ1000_DM_400_X_440_gen2_customizecards.dat
while others are not, like
bin/MadGraph5_aMCatNLO/Template/_customizecards.dat
I hope is clear.......

**Comment by abdollah110** [comment]
Adding all signal mass points

**Comment by perrozzi** [comment]
sorry but once more we didn't understand each other...
we strongly recommend to only update one example mass datacard for each type, and a script to produce the other mass points. but the script needs to be in a sensitive directory, i.e. for instance in the same directory as the example datacard...

**Comment by abdollah110** [comment]
So then it was what I committed initially. In order to make extra mass points, I need to have a script + a directory called Template. But you complained about the naming convention in the /Template directory (which basically can be any arbitrary names)
Am I still missing something?
If yes, maybe it is better to quickly chat rather exchange messages. I am at CERN

**Comment by perrozzi** [comment]
I don't think is appropriate to choose `bin/MadGraph5_aMCatNLO/Template/_customizecards.dat` for the template files. Please use `bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/DarkMatter_MonoLQ2/Template/_customizecards.dat` for instance

**Comment by abdollah110** [comment]
Auh, I see your point now. That directory was committed by mistake. I have just removed that.
Now I guess the last question is as following:
Would you like me to have various directories for various mass points (as it is now) or do you want me to switch back to have one example + the scripts + other stuff to generate many mass points' card?

**Comment by perrozzi** [comment]
> switch back to have one example + the scripts + other stuff to generate many mass points' card

thanks

**Comment by abdollah110** [comment]
Here is hopefully the last commit.
Can you merge this PR and we proceed with creating the gridpack??

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/DarkMatter_MonoLQ2/Template/_proc_card.dat [hunk 5]
```diff
@@ -0,0 +1,24 @@
+set group_subprocesses Auto
+set ignore_six_quark_processes False
+set loop_optimized_output True
+set loop_color_flows False
+set gauge unitary
+set complex_mass_scheme False
+set max_npoint_for_channel 0
+
+define l+ = e+ mu+
```
Comment (perrozzi) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/DarkMatter_MonoLQ2/Template/_proc_card.dat]: it looks to me that both the leptons and neutrinos are defined but not used, can you please cleanup?

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/DarkMatter_MonoLQ2/Template/_run_card.dat [hunk 6]
```diff
@@ -0,0 +1,282 @@
...
+# Store info for systematics studies                                 *
+# WARNING: If use_syst is T, matched Pythia output is                *
+#          meaningful ONLY if plotted taking matchscale              *
+#          reweighting into account!                                 *
+#*********************************************************************
+   True  = use_syst      ! Enable systematics studies
+#
+#**************************************                                  
+#
+0.5 1 2 = sys_scalefact  # factorization/renormalization scale factor
```
Comment (perrozzi) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/DarkMatter_MonoLQ2/Template/_run_card.dat]: @kdlong @qliphy I guess this section is related to syscalc and can be removed, can you please confirm?

Comment (abdollah110) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/DarkMatter_MonoLQ2/Template/_run_card.dat]: @kdlong @qliphy Can you please let me know what to do with these lines?

**Comment by abdollah110** [comment]
That's hopefully the last change unless something is critically WRONG in the card
we need to quickly proceed as this discussion is delaying the production procedure
